tokio/src/runtime/metrics/batch.rs

```diff
@@ -186,13 +208,19 @@ impl MetricsBatch {
 
     /// Start processing a batch of tasks
     pub(crate) fn start_processing_scheduled_tasks(&mut self) {
-        self.processing_scheduled_tasks_started_at = Instant::now();
+        if self.processing_scheduled_tasks_started_at.is_some() {
+            self.processing_scheduled_tasks_started_at = Some(Instant::now());
+        }
     }
 
     /// Stop processing a batch of tasks
     pub(crate) fn end_processing_scheduled_tasks(&mut self) {
-        let busy_duration = self.processing_scheduled_tasks_started_at.elapsed();
-        self.busy_duration_total += duration_as_u64(busy_duration);
+        if let Some(processing_scheduled_tasks_started_at) =
+            self.processing_scheduled_tasks_started_at
+        {
+            let busy_duration = processing_scheduled_tasks_started_at.elapsed();
```

Comment:
I like the impact of making this optional since it ensures we can't call `elapsed()` unless we actually got an instant.